src/main/java/org/folio/circulation/support/MultipleRecordFetcher.java

```diff
@@ -50,8 +93,20 @@ public MultipleRecordFetcher(
 
   public CompletableFuture<Result<MultipleRecords<T>>> findByQuery(
     Result<CqlQuery> queryResult) {
+    return findByQuery(queryResult, null);
+  }
 
-    return queryResult.after(query -> client.getMany(query, null))
+  private CompletableFuture<Result<MultipleRecords<T>>> findByQuery(
+    Result<CqlQuery> queryResult, Integer limit) {
+    return queryResult.after(query -> client.getMany(query, limit))
       .thenApply(result -> result.next(this::mapToRecords));
   }
+
+  private Result<MultipleRecords<T>> aggregate(List<MultipleRecords<T>> results) {
+    final Integer totalRecords = results.get(0).getTotalRecords();
+    final List<T> wrappedRecords = results.stream()
+      .flatMap(r -> r.getRecords().stream())
+      .collect(Collectors.toList());
+    return succeeded(new MultipleRecords<>(wrappedRecords, totalRecords));
```

Comment:
To maintain this abstraction (at least a little), should the total records not be the sum of all of the total records for all of the responses?

(I think this is a indicator that the multiple records abstraction might not fit in this context any more, any needs separating into two separate classes, one for when the total records matters and another where it doesn’t, but that may need to wait)

Reply:
Correct. I will update to sum the total records.